install/helm/agones/templates/crds/_gameserverspecschema.yaml

```diff
@@ -153,6 +153,42 @@ properties:
             type: integer
             title: The initial player capacity of this Game Server
             minimum: 0
+{{- if .featureCountsAndLists }}
+      counters:
+        type: array
```

Comment:
This should be an `object`

Reply:
A good example is `labels` in ObjectMeta:

https://github.com/googleforgames/agones/blob/921b1c1aef91d3fea08876113ecf1338cbae7fa5/install/helm/agones/templates/crds/k8s/_io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta.yaml#L68-L72

Reply:
Found some more references in the Open API Spec:
https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.0.md#parameter-object-examples

Particularly under _A free-form query parameter, allowing undefined parameters of a specific type:_

https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.0.md#model-with-mapdictionary-properties

(Although you can't use $ref in K8s CRDs, you have to specify everything inline, hence the need for includes).

That should help!

Reply:
Fixes in new PR https://github.com/googleforgames/agones/pull/2965